WowPacketParserModule.V8_0_1_27101/Parsers/QuestHandler.cs

```diff
@@ -0,0 +1,569 @@
+﻿using WowPacketParser.Enums;
+using WowPacketParser.Loading;
+using WowPacketParser.Misc;
+using WowPacketParser.Parsing;
+using WowPacketParser.Store;
+using WowPacketParser.Store.Objects;
+using WowPacketParserModule.V8_0_1_27101.Enums;
+
+namespace WowPacketParserModule.V8_0_1_27101.Parsers
+{
+    public static class QuestHandler
+    {
+        public static void ReadQuestRewardItemChoice(Packet packet, params object[] idx)
+        {
+            packet.ReadUInt32("ItemID", idx);
+            packet.ReadUInt32("UnkUint32_11", idx);
+            packet.ReadUInt32("UnkUint32_12", idx);
+
+            packet.ResetBitReader();
+
+            bool hasBonusData = packet.ReadBit("HasItemBonusData", idx);
+            bool hasString = packet.ReadBit("HasUnkString", idx);
+
+            if (hasBonusData)
+            {
+                packet.ReadByte("Context", idx);
+                int count = packet.ReadInt32("BonusIdCount", idx);
+
+                for (var j = 0; j < count; ++j)
+                    packet.ReadUInt32("BonusId", idx, j);
+            }
+
+            if (hasString)
+            {
+                int length = packet.ReadInt32();
+                packet.ReadWoWString("UnkString", length, idx);
+            }
+
+            packet.ReadUInt32("Quantity", idx);
+        }
+
+        public static void ReadQuestRewards(Packet packet, params object[] idx)
+        {
+            packet.ReadUInt32("ChoiceItemCount", idx);
+            packet.ReadUInt32("ItemCount", idx);
+
+            for (var i = 0; i < 4; ++i)
+            {
+                packet.ReadInt32("ID", idx, i);
+                packet.ReadInt32("Quantity", idx, i);
+            }
+
+            packet.ReadUInt32("RewardMoney", idx);
+            packet.ReadUInt32("XP", idx);
+            packet.ReadUInt64("UnkUInt64_HeartOfAzeroth", idx);
```

Comment:
ArtifactXP and ArtifactCategoryID

Reply:
Since it is not Called Artifact anymore it would prolly be named HeartOfAzerothXP <.<

Reply:
handt extracted bfa db2 files but im sure HoA data still in Artifect*.db2 💃